src/test/java/io/confluent/kafkarest/integration/ConsumerJsonTest.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.confluent.kafkarest.integration;
+
+import io.confluent.kafkarest.Versions;
+import io.confluent.kafkarest.entities.EmbeddedFormat;
+import io.confluent.kafkarest.entities.JsonConsumerRecord;
+import kafka.utils.TestUtils;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.junit.Before;
+import org.junit.Test;
+import scala.collection.JavaConversions;
+
+import javax.ws.rs.core.GenericType;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+public class ConsumerJsonTest extends AbstractConsumerTest {
+
+  private static final String topicName = "topic1";
+  private static final String groupName = "testconsumergroup";
+
+  private final List<ProducerRecord<Object, Object>> recordsWithKeys = Arrays.asList(
+      new ProducerRecord<Object, Object>(topicName, "key", "value"),
```

Comment:
Should we test anything besides strings? At least covering a few types with complex structure in one test would be a good idea, e.g. make sure integers, floats, strings, arrays, boolean, and null as fields in an object work.